File: 2002/ALL VB .NET SAMPLES/VB.NET - NET Framework - Create and use Trace Listeners/HTMLTraceListener.vb
Option Strict On

Imports System
Imports System.IO

' This class was created to output trace messages to an HTML file using tags

Public Class HTMLTraceListener
    Inherits System.Diagnostics.TextWriterTraceListener

    ' Use constructor from TextWriterTraceListener to write to a stream
    Public Sub New(ByVal stream As Stream)
        MyBase.New(stream)
    End Sub

    Public Sub WriteHeader(ByVal Title As String)
        ' New method to add a title to the top of the HTML document
        Writer.WriteLine("<head>")
        Writer.WriteLine("<title>" & Title & "</title>")
        Writer.WriteLine("</head>")
        Writer.WriteLine("<H2>" & Title & "</H2>")
        Writer.WriteLine("<P><HR>")

    End Sub

    Protected Overloads Overrides Sub WriteIndent()
        ' Override WriteIndent to handle indention in an HTML document
        ' using &nbsp; for a space
        Dim i As Integer
        Dim j As Integer

        If IndentLevel > 0 Then
            For i = 1 To IndentLevel
                For j = 1 To IndentSize
                    Writer.Write("&nbsp;")
                Next j
            Next i
        End If

    End Sub

    Public Overloads Overrides Sub WriteLine(ByVal message As String)
        ' Override WriteLine Method to add tags to output message
        Writer.Write("<B>" & Now() & " - ")

        If NeedIndent Then
            WriteIndent()
        End If

        Writer.WriteLine(message & "</B><BR>")
    End Sub

End Class

File: 2003/Code Samples/NET Framework/NET Framework - Create and use Trace Listeners/HTMLTraceListener.cs
using System;
using System.IO;

// This class was created to output trace messages to an HTML file using tags

public class HTMLTraceListener: System.Diagnostics.TextWriterTraceListener
{
    // Use constructor from TextWriterTraceListener to write to a stream
    public HTMLTraceListener(Stream stream): base(stream) {}

    public void WriteHeader(string Title)
	{
        // new method to add a title to the top of the HTML document
        Writer.WriteLine("<head>");
        Writer.WriteLine("<title>" + Title + "</title>");
        Writer.WriteLine("</head>");
        Writer.WriteLine("<H2>" + Title + "</H2>");
        Writer.WriteLine("<P><HR>");
    }

    protected override void WriteIndent()
	{
        // Override WriteIndent to handle indention in an HTML document
        // using &nbsp; for a space
		if (IndentLevel > 0)
		{
			for(int i = 1; i < IndentLevel; i++)
			{
				for(int j = 1; j < IndentSize; j++)
				{
					Writer.Write("&nbsp;");
				}
			}
        }
    }

    public override void WriteLine(string message)
	{
        // Override WriteLine Method to add tags to output message
        Writer.Write("<B>" + DateTime.Now.ToString() + " - ");

        if (NeedIndent)
		{
            WriteIndent();
        }

        Writer.WriteLine(message + "</B><BR>");
    }
}